packages/@lwc/template-compiler/src/shared/types.ts

```diff
@@ -25,9 +25,14 @@ export enum LWCDirectiveRenderMode {
     light = 'light',
 }
 
+type Metadata = { [key: string]: any };
+
 export interface BaseNode {
     type: string;
     location: SourceLocation;
+    // Meta is intended to be used internally only.
+    // The data in meta should not be exposed externally.
+    meta?: Metadata;
```

Comment:
Based on [feedback](https://github.com/salesforce/lwc/pull/4056#discussion_r1525331628), I intended to just put the part token on the `Attribute` / `Text` nodes but I figured we might want to have a more general property on the nodes so I added this field as a more broad place to attach metadata.